packages/oui-button/src/button.controller.js

```diff
@@ -0,0 +1,26 @@
+export default class {
+  constructor ($attrs, $element) {
+    'ngInject'
+
+    this.$attrs = $attrs
+    this.$element = $element
+  }
+
+  $onInit () {
+    if (angular.isUndefined(this.type)) {
+      this.type = 'button'
+    }
+
+    if (angular.isUndefined(this.variant)) {
+      this.variant = 'secondary'
+    }
+
+    this.disabled = angular.isDefined(this.$attrs.disabled) && angular.isUndefined(this.disabled)
```

Comment:
If the user does `<oui-button disabled="$ctrl.test"></oui-button>`, where `$ctrl.test` is `true`, are you sure it will not overwrite the parent value with `false`?

Reply:
Yes, i think so... I'll put back in a `if`.